public/static/plugins/perf/js/perf.js

```diff
@@ -188,10 +188,14 @@ mciModule.controller('PerfController', function PerfController(
       var key = tests[i];
       var series = trendSamples.seriesByName[key];
       var containerId = 'perf-trendchart-' + cleanId(taskId) + '-' + i;
+      var cps = scope.changePoints
+      var bfs = scope.buildFailures
 
       DrawPerfTrendChart({
         series: series,
-        changePoints: scope.changePoints[key],
+        // Concat orfaned change points and build failures
+        changePoints: (cps[key] || []).concat(cps[undefined] || []),
```

Comment:
I use `undefined` as a key because `_.groupBy` puts all non-grouping items into `undefined` group. There are nothing criminal in my opinion, since js object doesn't allow arbitrary key types. All keys are just strings. `undefined` must be cast to `"undefined"`.